runtime/binding-grpc/src/main/java/io/aklivity/zilla/runtime/binding/grpc/internal/config/GrpcBindingConfig.java

```diff
@@ -58,26 +71,41 @@ public final class GrpcBindingConfig
     private static final byte[] BIN_SUFFIX = "-bin".getBytes();
     private final HttpGrpcHeaderHelper helper;
 
+    private final Long2ObjectHashMap<CatalogHandler> handlersById;
+    private final Int2ObjectHashMap<GrpcProtobufConfig> protobufsBySchemaId;
+    private final List<CatalogedConfig> catalogs;
+    private final Int2IntHashMap schemaIds;
+
+    public final List<GrpcRouteConfig> routes;
     public final long id;
     public final String name;
     public final KindConfig kind;
     public final GrpcOptionsConfig options;
-    public final List<GrpcRouteConfig> routes;
 
 
     public GrpcBindingConfig(
         BindingConfig binding,
-        MutableDirectBuffer metadataBuffer)
+        MutableDirectBuffer metadataBuffer,
+        ToLongFunction<String> resolveId,
+        Function<Long, CatalogHandler> supplyCatalog)
     {
         this.id = binding.id;
         this.name = binding.name;
         this.kind = binding.kind;
         this.options = GrpcOptionsConfig.class.cast(binding.options);
         this.routes = binding.routes.stream().map(GrpcRouteConfig::new).collect(toList());
         this.helper = new HttpGrpcHeaderHelper(metadataBuffer);
+        this.schemaIds = new Int2IntHashMap(0);
+        this.handlersById = new Long2ObjectHashMap<>();
+        this.protobufsBySchemaId = new Int2ObjectHashMap<>();
+        this.catalogs = binding.catalogRef != null ? binding.catalogRef.catalogs.stream().map(c ->
+        {
+            c.id = resolveId.applyAsLong(c.name);
```

Comment:
Setting the namespaced id on the `CatalogedConfig` should be handled already by the engine, as it is for all the other generic config objects.

Then you don't need to do this part here and may not need `resolveId` at all.